controllers/articlesController.js

```diff
@@ -93,4 +82,154 @@ export default class PostController {
       });
     }
   }
+
+  /**
+   * @description It helps the user to fetch all of the created articles.
+   * @param  {object} req - The request object
+   * @param  {object} res - The response object
+   * @returns {object} It returns the request's response object
+   */
+
+  static async findAll(req, res, next) {
+    const { user: { id = null } = {} } = req;
+    const { author, favorited, tag } = req.query;
+    const { page = 1, limit = 20 } = req.query;
+    try {
+      const all = await Article.findAll({
+        offset: (Number(page) - 1) * Number(limit),
+        limit,
+        order: [["createdAt", "DESC"]],
+        where: { deletedAt: null },
+        attributes: { exclude: ["deletedAt"] },
+        include: [
+          {
+            model: User,
+            as: "author",
+            attributes: ["username", "bio", "image"],
+            where: author ? { username: author } : {},
+            include: [
+              { model: User, as: "followers", attributes: ["id", "username"] }
+            ]
+          },
+          { model: User, as: "likes", attributes: ["username"], through: {} },
+          { model: Tag, attributes: ["name"], as: "tagsList", through: {} }
+        ]
+      });
+      if (all.length) {
+        const articles = all.map(each => {
+          const { followers, username, bio, image } = each.get().author;
+          const { tagsList, likes, ...article } = each.get();
+          const following = !!_.dropWhile(followers, f => f.id !== id).length;
+          return {
+            ...article,
+            likes: likes.map(like => like.username),
```

Comment:
@Musigwa Article's likes information should be returned separately. Users only need the see the metadata from articles' list. You can just return the number of likes as a counter

Reply:
@esuka, you can navigate through our project's UI pages to explore more. For us, we will fetch the articles once so that if a user clicks on any of it from the list, the first page will pass the data to the next `single article page` as we will use React.Js.
Here are our [UI pages](http://github.andela.com/titan-dev-ah-frontend/pages/landing.html)